new_smart_launch.sh

```diff
@@ -10,15 +10,41 @@ echo '  \  \::/     \  \::/       \  \:\  /:/  \__\/  \:\   \  \:\  /:/   \  \::
 echo '   \  \:\      \  \:\        \  \:\/:/        \  \:\   \  \:\/:/     \  \:\          /__/:/    \  \:\/:/  '
 echo '    \  \:\      \  \:\        \  \::/          \__\/    \  \::/       \  \:\         \__\/      \  \::/   '
 echo '     \__\/       \__\/         \__\/                     \__\/         \__\/                     \__\/    '
-# Checking if server is ready 
-if [ $FACTORIO_WAITING == true ] 
-then 
-  until [ -f /opt/factorio/saves/ready ] 
-  do 
-    echo "# Waiting for backup daemon to be ready" 
-    sleep 1 
-  done 
+# Checking if server is ready
+if [ $FACTORIO_WAITING == true ]
+then
+  until [ -f /opt/factorio/saves/ready ]
+  do
+    echo "# Waiting for backup daemon to be ready"
+    sleep 1
+  done
 fi
+# Populate server-settings.json
+SERVER_SETTINGS=/opt/factorio/server-settings.json
+cat << EOF > $SERVER_SETTINGS
+{
+  "name": "$FACTORIO_SERVER_NAME",
+  "description": "$FACTORIO_SERVER_DESCRIPTION",
+  "max_players": "$FACTORIO_SERVER_MAX_PLAYERS",
+
+  "_comment_visibility": ["public: Game will be published on the official Factorio matching server",
+                          "lan: Game will be broadcast on LAN",
+                          "hidden: Game will not be published anywhere"],
+  "visibility": "$FACTORIO_SERVER_VISIBILITY",
```

Comment:
Might want to do a preemptive check for public without a username / password and default back to "hidden", maybe also alert the user that this is the reason its happening.